spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -209,14 +213,35 @@ public long executePartitionedDmlStatement(Statement statement) {
 		return entities;
 	}
 
-	private <T> boolean isEligibleForEagerFetch(KeySet keys, SpannerReadOptions options,
-			SpannerPersistentEntity<T> persistentEntity) {
-		//an entity is eligible if all of the following true:
-		//1. entity has eager loaded properties
-		//2. there are no read options, as they can't be applied to a query
-		//3. key set does not have ranges, as they can't be used in a query
-		return persistentEntity.hasEagerlyLoadedProperties() &&
-				options == null && !keys.getRanges().iterator().hasNext();
+	/**
+	 * In many cases {@link KeySet} with {@link SpannerReadOptions} are compatible with
+	 * {@link SpannerReadOptions}. The method throws exception when it is impossible.
+	 * @param options read-parameters
+	 * @return query-parameters
+	 * @throws IllegalArgumentException when {@link SpannerQueryOptions} can't be converted to {@link SpannerQueryOptions}
+	 * 	or {@code keys} have "ranges".
+	 */
+	private static SpannerQueryOptions toQueryOption(KeySet keys, SpannerReadOptions options) throws IllegalArgumentException {
```

Comment:
Move this method to `SpannerQueryOptions` class and rename to `fromReadOptions`

Reply:
Another option is to move this method to `SpannerReadOptions`, so you could call it like this:
`options.toQueryOptions()`

Reply:
Ah, well, clear